exp/controllers/awsmachinepool_controller.go

```diff
@@ -531,11 +532,26 @@ func asgNeedsUpdates(machinePoolScope *scope.MachinePoolScope, existingASG *expi
 		return true
 	}
 
+	if checkAZsNeedsUpdate(machinePoolScope.AvailabilityZones(), existingASG.AvailabilityZones) {
+		return true
+	}
+
 	// todo subnet diff
 
 	return false
 }
 
+func checkAZsNeedsUpdate(input, existing []string) bool {
+	sort.Slice(input, func(i, j int) bool {
+		return input[i] < input[j]
+	})
+	sort.Slice(existing, func(i, j int) bool {
+		return existing[i] < existing[j]
+	})
+
+	return reflect.DeepEqual(input, existing)
```

Comment:
Shouldn't this be `!reflect.DeepEqual(input, existing)`, coz if both are same we won't need the update?

Reply:
Yes, good spot @Ankitasw